crates/rattler_lock/src/lib.rs

```diff
@@ -219,18 +218,18 @@ impl LockFile {
 
     /// Returns an iterator over all environments defined in the lock-file.
     pub fn environments(&self) -> impl ExactSizeIterator<Item = (&str, Environment<'_>)> + '_ {
-        self.inner
-            .environment_lookup
-            .iter()
-            .map(move |(name, index)| {
-                (
-                    name.as_str(),
-                    Environment {
-                        lock_file: self,
-                        index: *index,
-                    },
-                )
-            })
+        let mut environments: Vec<_> = self.inner.environment_lookup.iter().collect();
+        environments.sort_by(|(a, _), (b, _)| a.cmp(b));
```

Comment:
This is not strictly needed for the function to work correctly. Rather it is something we want in the test. I think we should modify the tests instead.